src/AutoRest.CSharp/build/CodeGeneration.targets

```diff
@@ -17,14 +17,15 @@
     <AutoRestSharedCodeDirectory>$(MSBuildThisFileDirectory)../content/Generator.Shared/</AutoRestSharedCodeDirectory>
     <ManagementSharedCodeDirectory>$(MSBuildThisFileDirectory)../content/Management.Shared/</ManagementSharedCodeDirectory>
 
-    <_GenerateCode Condition="'$(AutoRestInput)' != ''">true</_GenerateCode>
+    <_GenerateCode Condition="'$(AutoRestInput)' != '' OR '$(CadlInput)' != ''">true</_GenerateCode>
     <UseDefaultNamespaceAndOutputFolder Condition="'$(UseDefaultNamespaceAndOutputFolder)' == ''">true</UseDefaultNamespaceAndOutputFolder>
     <_AutoRestCommand>node $(AutoRestEntryPoint) --max-memory-size=8192 --skip-csproj --skip-upgrade-check --version=$(AutoRestCoreVersion) $(AutoRestInput) $(AutoRestAdditionalParameters) --use=$(MSBuildThisFileDirectory)../tools/netcoreapp3.1/any/ --clear-output-folder=true --shared-source-folders=&quot;$(AzureCoreSharedCodeDirectory);$(AutoRestSharedCodeDirectory)&quot;</_AutoRestCommand>
     <_AutoRestCommand Condition="'$(UseDefaultNamespaceAndOutputFolder)' == 'true'">$(_AutoRestCommand) --output-folder=$(MSBuildProjectDirectory)/Generated --namespace=$(RootNamespace)</_AutoRestCommand>
+    <_CadlGenerateCommand Condition="'$(CadlInput)' != ''">npx cadl compile --emit @azure-tools/cadl-csharp --output-path $(MSBuildProjectDirectory)/Generated --option @azure-tools/cadl-csharp.clear-output-folder=true $(CadlInput)</_CadlGenerateCommand>
   </PropertyGroup>
 
-  <Target Name="GenerateCode" Condition="'$(_GenerateCode)' == 'true'" >
-    <ReadLinesFromFile File="$(AutoRestInput)">
+  <Target Name="GenerateCode" Condition="'$(_GenerateCode)' == 'true' AND Exists('$(AutoRestInput)')" >
+    <ReadLinesFromFile File="$(AutoRestInput)" Condition="Exists('$(AutoRestInput)')">
```

Comment:
Is this already checked in the previous line?

Reply:
Yes, it can be remove. removed